### DISCUSSION TITLE
I don't understand storage and snapshotting

### DISCUSSION THREAD

**Discussion by th** [go-shiori/shiori#661] [Q&A]
## Data
- **Shiori version**

Don't know. There is no version subcommand or --version flag.

- **Database Engine**: 

SQLite, I guess?

- **Operating system**: 

Windows 11

- **CLI/Web interface/Web Extension**: 

CLI and web

## Describe the bug / actual behavior
I import bookmarks from a Netscape HTML file using shiori import --portable. Then I do shiori serve --portable.

No archive content is created. That is weird, since I have automatic archive creation ON in the web interface options.

I do shiori update, and confirm. I'm not sure if "update      Update the saved bookmarks" is what I want, since I don't want to update, I want to snapshot for the first time, but I try that.

Shiori updates/snapshots all bookmarks (and the wording is surprising, I would expect to use update for re-snapshotting a link, not for getting a first archive version).

Later I import some more bookmarks.

Now the conundrum: I don't really want to throw away the old  archives (so shiori update and update all is out), but there is no other, incremental way to update the archives, is there?

I also don't know what "Refresh storage" in the web interface does (nothing, as far as I can observe).

## Expected behavior
Some documented way to (a) snapshot links and (b) then incrementally snapshot new links.

Also, maybe you could settle on "archive" for the "storage" thing, because the latter term is pretty unclear to the user. I assumed at first it's about the link title and metadata in the database, not the content.

## To Reproduce
See above.

**Comment by fmartingr** [comment]
Hello @th , 

I'm going to answer step by step, to ease up all issues:

> since I have automatic archive creation ON in the web interface options.

Archive creation in the web interface is for newly added archives, not for imports.

> Shiori updates/snapshots all bookmarks (and the wording is surprising, I would expect to use update for re-snapshotting a link, not for getting a first archive version).

The `update` command updates a bookmark in different ways: You can change attributes (title, excerpt, ..), the readable version and the archive version (the snapshot as you say). Maybe wording can be improved as you say.

> I also don't know what "Refresh storage" in the web interface does (nothing, as far as I can observe).

The button is just a refresh, in case any change happened while the web page is opened.

> Some documented way to (a) snapshot links and (b) then incrementally snapshot new links.

Do you mean via CLI or via web interface? since it seems you mixed some comments? Any suggestions welcome of course. It would help knowing the specific flows you followed into more detail, if you can.

**Comment by th** [comment]
The use case is this: I‘m exporting my raindrop.io bookmarks as  Netscape HTML and do a shiori import.

A few weeks later I re-export from raindrop.io. Can I incrementally update shiori?

**Comment by fmartingr** [comment]
> The use case is this: I‘m exporting my raindrop.io bookmarks as Netscape HTML and do a shiori import.
> 
> A few weeks later I re-export from raindrop.io. Can I incrementally update shiori?

The `import` command does not remove old bookmarks, it will check if the bookmarks you are importing are already on the database, so you should be able to run import multiple times without issues.

**Reply by th** [reply]
As far as I can see, the only way to get snapshots is shiori update, and that only offers update all or abort. But I think old snapshots should be kept alone (either because re-snapshotting takes time, or because the links may be dead by now), only newly imported stuff shoul be snapshotted.
